core/system/inc/page_allocator_config.h

```diff
@@ -42,4 +42,27 @@ typedef uint8_t page_owner_t;
 /* Define a unused value for the page table. */
 #define UVISOR_PAGE_UNUSED ((page_owner_t) (-1))
 
+/** Sets the page bit in the page map array.
+ * @param map   an array of `uint32_t` containing the page map
+ * @param page  the index of the page to be set
+ */
+#define UVISOR_PAGE_MAP_SET(map, page) \
+    (map)[(page) / 32] |= (1 << ((page) & 31))
+
+/** Clears the page bit in the page map array.
+ * @param map   an array of `uint32_t` containing the page map
+ * @param page  the index of the page to be set
+ */
+#define UVISOR_PAGE_MAP_CLEAR(map, page) \
+    (map)[(page) / 32] &= ~(1 << ((page) & 31))
+
+/** Check if the page bit is set int the page map array.
+ * @param map   an array of `uint32_t` containing the page map
+ * @param page  the index of the page to be set
+ * @retval 0    if page bit is not set
+ * @retval !0   if page bit is set
+ */
+#define UVISOR_PAGE_MAP_GET(map, page) \
+    ((map)[(page) / 32] & (1 << ((page) & 31)))
+
```

Comment:
@niklas: I would consider making these functions static inline and move them to the C file.